### PR TITLE
dbsync drop if needed

### LINKED ISSUE #474: add drop_if_error to Parsons DBSync
I had a DBSync set to `truncate` and it failed because of a varchar length or something equally benign. In this case, I am also more than happy to drop when needed. Seems like we could easily add a `try/except` block where the parameter option is `drop_if_error`.

### PR DESCRIPTION
**Pull request by elyse-weiss**
Closes #474

### CODE REVIEW

File: parsons/databases/db_sync.py [hunk 1]
```diff
@@ -56,9 +58,13 @@ def table_sync_full(self, source_table, destination_table, if_exists='drop',
...
             if if_exists == 'drop':
                 destination_tbl.drop()
-            elif if_exists == 'truncate':
-                self._check_column_match(source_tbl, destination_tbl)
-                destination_tbl.truncate()
+            elif if_exists in ('truncate','drop_if_needed'):
+                try:
+                    self._check_column_match(source_tbl, destination_tbl)
+                    destination_tbl.truncate()
+                except:
```
Comment (SorenSpicknall) [inline parsons/databases/db_sync.py:65]: Doesn't this except clause make `truncate` and `drop_if_needed` exactly the same? i.e. I'd only do this for `drop_if_needed`, and allow `truncate` to fail if truncation doesn't work.

**Review by SorenSpicknall — CHANGES_REQUESTED** [review]
One note on logic. Also, might be worth it to change your description to link an issue, since you're accidentally saying that this PR closes this PR and linking to the PR itself.

File: parsons/databases/db_sync.py (hunk 1 above)
Comment (elyse-weiss) [reply parsons/databases/db_sync.py:65]: You are right! Fixing!

**Review by SorenSpicknall — APPROVED** [review]
Looks ready to go. Unrelated tests are failing due a known Zoom connector bug, but this functionality is passing.

**Pull request merged**